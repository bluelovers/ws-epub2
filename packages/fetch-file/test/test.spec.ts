/**
 * Created by user on 2020/3/28.
 */

import fetchFileOrUrl from '../index';

describe('test fetch', function ()
{

	it('fetch url', async () => {

		let ret = await fetchFileOrUrl({
			url: 'https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_92x30dp.png',
		})
			.tap(fo =>
			{
				console.dir(fo)
			})
		;

		expect(ret).toHaveProperty('url');
		expect(ret).toHaveProperty('mime');
		expect(ret).toHaveProperty('ext');
		expect(ret).toHaveProperty('name');
		expect(ret).toHaveProperty('data');
		expect(ret).toMatchSnapshot({
			data: expect.any(Buffer),
		});

	})

});
