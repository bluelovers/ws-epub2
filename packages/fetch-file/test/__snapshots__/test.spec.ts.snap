// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test fetch fetch url 1`] = `
{
  "data": Any<Buffer>,
  "ext": ".webp",
  "mime": "image/png",
  "name": "0ccb6d7c.webp",
  "url": "https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_92x30dp.png",
}
`;
